.github/workflows/ci.yml

```diff
@@ -42,137 +47,94 @@ jobs:
-      - name: Run scenario against image
-        run: ./temporal-omes run-scenario --scenario workflow_with_single_noop_activity --log-level debug --server-address 127.0.0.1:10233 --run-id ${{ github.run_id }} --connect-timeout 1m --iterations 5
+          ./scripts/install-java.sh
+          ./scripts/install-go.sh
+      - name: Lint Java
+        run: |
+          ./scripts/lint-java-worker.sh
+      - name: Check Java formatting
+        run: |
+          git diff --exit-code || (echo "Java worker has uncommitted formatting changes after linting" && exit 1)
+      - name: Build Java worker
+        run: |
+          ./scripts/build-java-worker.sh
 
   build-lint-test-python:
     runs-on: ubuntu-latest
     steps:
       - name: Print build information
         run: "echo head_ref: ${{ github.head_ref }}, ref: ${{ github.ref }}"
       - name: Checkout repo
-        uses: actions/checkout@v2
-      - name: Load versions
+        uses: actions/checkout@v4
+      - name: Install mise
+        uses: jdx/mise-action@v2
+      - name: Install Tools
         run: |
-          set -a && source versions.env && set +a
-          printenv >> $GITHUB_ENV
-      - name: Setup Python
-        uses: actions/setup-python@v1
-        with:
-          python-version: "${{ env.PYTHON_VERSION }}"
-      - name: Install uv
-        uses: astral-sh/setup-uv@v5
-      - name: Install poe
-        run: uv tool install poethepoet
-      - name: Prepare Python worker environment
-        run: cd workers/python && uv sync
-      - name: Lint Python worker
-        run: cd workers/python && poe lint
-      - name: Setup Go
-        uses: actions/setup-go@v2
-        with:
-          go-version: "${{ needs.load-versions.outputs.go_version }}"
-      - name: Build exe
-        run: go build -o temporal-omes ./cmd
-      - name: Run local scenario with worker
-        run: ./temporal-omes run-scenario-with-worker --scenario workflow_with_single_noop_activity --log-level debug --language python --embedded-server --iterations 5 --version ${{ env.PYTHON_SDK_VERSION }}
-      - name: Build worker image
-        run: ./temporal-omes build-worker-image --language python --version ${{ env.PYTHON_SDK_VERSION }} --tag-as-latest
-      - name: Run worker image
-        run: docker run --rm --detach -i -p 10233:10233 omes:python-${{ env.PYTHON_SDK_VERSION }} --scenario workflow_with_single_noop_activity --log-level debug --language python --run-id ${{ github.run_id }} --embedded-server-address 0.0.0.0:10233
-      - name: Run scenario against image
-        run: ./temporal-omes run-scenario --scenario workflow_with_single_noop_activity --log-level debug --server-address 127.0.0.1:10233 --run-id ${{ github.run_id }} --connect-timeout 1m --iterations 5
+          ./scripts/install-python.sh
+          ./scripts/install-go.sh
```

Comment:
If this is leaning too much on mise, I can also revert this.

While it is an extra dependency, it is one that is very easy to remove again. And with this, we also test the install scripts, too.